tools/ci/java-ci-tools/src/main/java/org/apache/flink/tools/ci/licensecheck/LicenseChecker.java

```diff
@@ -0,0 +1,320 @@
+
+	// pattern for maven shade plugin
+	private static final Pattern SHADE_NEXT_MODULE_PATTERN = Pattern.compile(".*:shade \\((shade-flink|default)\\) @ ([^ _]+)(_[0-9.]+)? --.*");
+	private static final Pattern SHADE_INCLUDE_MODULE_PATTERN = Pattern.compile(".*Including ([^:]+):([^:]+):jar:([^ ]+) in the shaded jar");
+
+	// pattern for maven-dependency-plugin copyied dependencies
+	private static final Pattern DEPENDENCY_COPY_NEXT_MODULE_PATTERN = Pattern.compile(".*maven-dependency-plugin:3.1.1:copy \\([^)]+\\) @ ([^ _]+)(_[0-9.]+)? --.*");
+	private static final Pattern DEPENDENCY_COPY_INCLUDE_MODULE_PATTERN = Pattern.compile(".*Configured Artifact: ([^:]+):([^:]+):([^:]+):jar.*");
+
+	private static final Pattern NOTICE_DEPENDENCY_PATTERN = Pattern.compile("- ([^ :]+):([^:]+):([^ ]+)($| )|.*bundles \"([^:]+):([^:]+):([^\"]+)\".*");
+
+	private int run(File buildResult, File root) throws IOException {
+		int severeIssueCount = 0;
+		// parse included dependencies from build output
+		Multimap<String, IncludedDependency> modulesWithShadedDependencies = parseModulesFromBuildResult(buildResult);
+		LOG.info("Extracted " + modulesWithShadedDependencies.asMap().keySet().size() + " modules with a total of " + modulesWithShadedDependencies.values().size() + " dependencies");
+
+		// find modules producing a shaded-jar
+		List<File> noticeFiles = findNoticeFiles(root);
+		LOG.info("Found {} NOTICE files to check", noticeFiles.size());
+
+		// check that all required NOTICE files exists
+		severeIssueCount += ensureRequiredNoticeFiles(modulesWithShadedDependencies, noticeFiles);
+
+		// check each NOTICE file
+		for (File noticeFile: noticeFiles) {
+			severeIssueCount += checkNoticeFile(modulesWithShadedDependencies, noticeFile);
+		}
+
+		// find modules included in flink-dist
+
+		return severeIssueCount;
+	}
+
+	private int ensureRequiredNoticeFiles(Multimap<String, IncludedDependency> modulesWithShadedDependencies, List<File> noticeFiles) {
+		int severeIssueCount = 0;
+		Set<String> shadingModules = new HashSet<>(modulesWithShadedDependencies.keys());
+		shadingModules.removeAll(noticeFiles.stream().map(LicenseChecker::getModuleFromNoticeFile).collect(Collectors.toList()));
+		for (String moduleWithoutNotice : shadingModules) {
+			if (!MODULES_SKIPPING_DEPLOYMENT.contains(moduleWithoutNotice)) {
+				LOG.warn("Module {} is missing a NOTICE file. It has shaded dependencies: {}", moduleWithoutNotice, modulesWithShadedDependencies.get(moduleWithoutNotice));
+				severeIssueCount++;
+			}
+		}
+		return severeIssueCount;
+	}
+
+	private static String getModuleFromNoticeFile(File noticeFile) {
+		File moduleFile = noticeFile.getParentFile() // META-INF
+			.getParentFile() // resources
+			.getParentFile() // main
+			.getParentFile() // src
+			.getParentFile(); // <-- module name
+		return moduleFile.getName();
```

Comment:
```suggestion
	private static String getModuleFromNoticeFile(Path noticeFile) {
		File moduleDirectory = noticeFile.getParent() // META-INF
			.getParent() // resources
			.getParent() // main
			.getParent() // src
			.getParent(); // <-- module name
		return moduleDirectory.getFileName();
```
With this change we should be able to use `Path` everywhere, so we can stop converting things everywhere.